### PR TITLE
Feat/add recipe to history

### PR DESCRIPTION
**Pull request by CataAlamos**
Se agrego la funcionalidad de agregar y eliminar recetas favoritas y el historial de recetas.

### CODE REVIEW

File: app/components/recipeCard.jsx [hunk 1]
```diff
@@ -31,9 +31,16 @@ export default function RecipeCard({ N, recipeName, imageUrl, recipeId }) {
...
   };
 
-  const handleConfirm = () => {
+  const handleConfirm = async () => {
     setShowAlert(false);
-    console.log("Confirmed 'Hacer Receta'");
+    try {
+      const response = await registerRecipeConsumption(recipeId);
+      console.log(response);
+      alert(`${response.message}`);
```
Comment (PipeOlivares) [inline app/components/recipeCard.jsx:39]: Is this alert showing text that the user will understand, or is it mainly for our debugging purposes?

File: app/routes/recipes-generator.jsx [hunk 2]
```diff
@@ -44,54 +36,209 @@ export default function RecipiesGenerator() {
...
+    }
+  };
+
   return (
     <div className="generalContainer flex">
       <SideBar />
       <div className="ContainerBody relative h-[1100px] grow bg-[#E5E9F0] p-[60px] z-[0] overflow-y-hidden">
-        <h1 className="text-3xl text-[#182F40] font-bold">
-          {headerText}<span className="text-7xl font-extralight">{headerSpanText}</span>
+        <h1 className="text-3xl font-bold text-[#182F40] font-kantumruy">
```
Comment (PipeOlivares) [inline app/routes/recipes-generator.jsx:139]: I think that this breaks the formatting requirements for mantaining a uniform font throught the project.
If the font-change is **necessary** and **part of the figma**, then ignore!

File: app/routes/recipes-generator.jsx [hunk 3]
```diff
@@ -44,54 +36,209 @@ export default function RecipiesGenerator() {
...
+                  toggleFavorite(recipeId);
+                }}
+              />
+            )}
           </div>
         </div>
-
-        {/* Background images and Logo */}
-        <img src={ellipseBackground} alt="elipse" className="absolute top-[-35%] left-[-10%] z-[-1]"/>
+         <img src={ellipseBackground} alt="elipse" className="absolute top-[-35%] left-[-10%] z-[-1]"/>
```
Comment (PipeOlivares) [inline app/routes/recipes-generator.jsx:236]: In this line, there might be an extra indentation which might break the HTML Syntax. 
Please review!

File: app/routes/useApi.jsx [hunk 4]
```diff
@@ -150,15 +204,16 @@ export default function useApi() {
     */
     const addIngredientsToPantry = async (ingredients) => {
         const body = { ingredients };
-        return await apiCall('/pantry/addIngredients', 'POST', body);
+        body.sign = 1;
```
Comment (PipeOlivares) [inline app/routes/useApi.jsx:207]: Is the `body.sign = -1` a change agreed-to with the backend team?

File: vite.config.js [hunk 5]
```diff
@@ -59,11 +58,38 @@ export default defineConfig({
   },
   server: {
     watch: {
-      usePolling: true,   // Activa polling para detectar cambios en entornos donde el file watching puede fallar
```
Comment (PipeOlivares) [inline vite.config.js:62]: It might be useful to leave these comments so that a future developer knows how to change the Vite configurations.

**Review by PipeOlivares — CHANGES_REQUESTED** [review]

File: app/routes/useApi.jsx (hunk 4 above)
Comment (CataAlamos) [reply app/routes/useApi.jsx:207]: En add Ingredient es 1 y en remove es -1

File: vite.config.js (hunk 5 above)
Comment (CataAlamos) [reply vite.config.js:62]: Agregue nuevamente el comentario

File: app/routes/recipes-generator.jsx (hunk 3 above)
Comment (CataAlamos) [reply app/routes/recipes-generator.jsx:236]: Arreglado.

File: app/routes/recipes-generator.jsx (hunk 2 above)
Comment (CataAlamos) [reply app/routes/recipes-generator.jsx:139]: Quito el font, debería ser automático.

**Review by PipeOlivares — APPROVED** [review]
The requested changes where implemented correctly, or ignored when necessary.

**Pull request merged**